src/index.ts

```diff
@@ -113,33 +151,45 @@ async function resolve(
   timestamp: number,
   config: NftResovlerConfig
 ): Promise<DIDResolutionResult> {
-  const asset = idToAsset(methodId)
-  const ethRpc = config.ethereumRpcs[ChainID.format(asset.chainId)]
-  const account = await assetToAccount(asset, timestamp, ethRpc)
-  const controller = await accountToDid(account, timestamp, config.ceramic)
-  return {
-    didResolutionMetadata: { contentType: DID_JSON },
-    didDocument: wrapDocument(did, account, controller),
-    didDocumentMetadata: {}
-  }
-}
+  const asset = idToAsset(methodId);
+  // for 1155s, there can be many accounts that own a single asset
+  const owningAccounts = await assetToAccount(asset, timestamp, config.subGraphUrls);
+  const controllers = await accountsToDids(owningAccounts, timestamp, config.ceramic);
+  const metadata: DIDDocumentMetadata = {};
 
-interface EthereumRpcEndpoints {
-  [chainId: string]: string
-}
+  // TODO create, update, versionId
```

Comment:
Agreed with `update` and `versionId`, though `create` may be useful for platforms looking to leverage the time at which the NFT was created: For example, the HashMasks each generate a certain amount of NameChangeTokens per day since its creation and if `create` is in the metadata, a platform wouldn't have to perform a query to their contract or subgraph.